web/src/app/app.component.ts

```diff
@@ -0,0 +1,160 @@
+declare let require: any;
+declare var web3: any;
+import { Component } from '@angular/core';
+const etherlime = require('etherlime');
+const ethers = require('ethers');
+const ToDo = require('../../../build/ToDoManager.json');
+
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+
+
+export class AppComponent {
+  title = 'My Decentralized App';
+
+  public successMessage: string;
+  public infoMessage: string;
+  public errorMessage: string;
+  public contractInstance: any;
+  public contractAddress = '0xc9707E1e496C12f1Fa83AFbbA8735DA697cdBf64';
+  public toDoStatus: any;
+  public inProgressStatus: any;
+  public doneStatus: any;
+  public inactiveButton;
+
+  constructor() {
+    this.toDoStatus = [];
+    this.inProgressStatus = [];
+    this.doneStatus = [];
+
+    etherlime.ContractAt(ToDo, this.contractAddress).then((result) => {
```

Comment:
In order to be able to use async/await you should do this logic in the ngOnInit @ochikov can you help?